core/completion.py

```diff
@@ -1212,14 +1217,23 @@ def _GetNextCompletion(self, state):
 
       comp = Api(line=buf, begin=begin, end=end)
 
-      self.comp_iter = self.root_comp.Matches(comp)
+      if mylib.PYTHON:
+        self.comp_iter = self.root_comp.Matches(comp)
+      else:
+        self.comp_iter = list(self.root_comp.Matches(comp))
 
     assert self.comp_iter is not None, self.comp_iter
 
-    try:
-      next_completion = self.comp_iter.next()
-    except StopIteration:
-      next_completion = None  # signals the end
+    if mylib.PYTHON:
+      try:
+        next_completion = self.comp_iter.next()
+      except StopIteration:
+        next_completion = None  # signals the end
+    else:
+      try:
+        next_completion = self.comp_iter.pop()
```

Comment:
I feel like this will have the reverse order of next()?   It pops from the end

But I guess we'll find out through tests

Reply:
oh good question. i'll do a quick test to check and update this if needed

Reply:
you were right! this should be `pop(0)`

Reply:
I think it's better to do `L.reverse()` and then `pop()`  

Because pop(0) does the memcpy() every time I think, which would give a quadratic iteration

And then reverse() might allow us to copy it from the stack to a GC list ??